### PR TITLE
docs(0002): Edits to Crypto-Conditions spec

### PR DESCRIPTION
**Pull request by mDuo13**
Reviewers: @justmoon @adrianhopebailie

### CODE REVIEW

File: 0002-crypto-conditions/0002-crypto-conditions.md [hunk 1]
```diff
@@ -92,78 +92,79 @@ Crypto-conditions provide a mechanism to describe a signed message such that mul
...
 
-## Terminology {#terminology}
-The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119].
+Fulfillments are **cryptographically verifiable messages** that prove an event occurred. If you transmit a fulfillment, then everyone who has the condition can agree that the condition has been met.
 
-Within this specification, the term "condition" refers to the hash of a description of a signed message.
+In the Interledger protocol, crypto-conditions and fulfillments provide irrepudiable proof that a transfer occurred in one ledger, as messages that can be easily shared with other ledgers. This allows ledgers escrow funds or hold a transfer conditionally, then execute the transfer automatically when the ledger sees the fulfillment of the stated condition.
```
Comment (adrianhopebailie) [inline 0002-crypto-conditions/0002-crypto-conditions.md]: Do we want to reference Interledger? I'd suggest we don't as we see CC having value beyond ILP

File: 0002-crypto-conditions/0002-crypto-conditions.md [hunk 2]
```diff
@@ -92,78 +92,79 @@ Crypto-conditions provide a mechanism to describe a signed message such that mul
...
 
 
 
-# Format {#format}
+# Format
 
-## Binary Encoding {#binary-encoding}
+## Binary Encoding
 
 An description of crypto-conditions is provided in this document using Abstract Syntax Notation One (ASN.1) as defined in [ITU.X680.2015](#itu.X680.2015). Implementations of this spec MUST support encoding and decoding using Octet Encoding Rules (OER) as defined in [ITU.X696.2015](#itu.X696.2015).
```
Comment (adrianhopebailie) [inline 0002-crypto-conditions/0002-crypto-conditions.md]: Change:
> An description
To:
> A description

File: 0002-crypto-conditions/0002-crypto-conditions.md [hunk 3]
```diff
@@ -232,61 +250,77 @@ type
...
+    cf:0:dGhpcyBpcyBzb21lIHRleHQgdGhhdCBtRHVvMTMgd3JvdGU
+
+
+The example has the following attributes:
+
+| Field                | Value | Description |
+|----------------------|-------|----------------------------------------------|
+| Preface              | `cf`  | Indicates this is a fulfillment. |
+| type                 | `0`   | Type 0 is [PREIMAGE-SHA-256][]. |
+| payload              | `dGhpcyBpcyBzb21lIHRleHQgdGhhdCBtRHVvMTMgd3JvdGU` | The BASE64URL-encoded SHA-256 preimage of the condition, since this is a PREIMAGE-SHA-256 type fulfillment. In this case, it is an arbitrary string. |
```
Comment (adrianhopebailie) [inline 0002-crypto-conditions/0002-crypto-conditions.md]: > "this is some text that mDuo13 wrote"

Don't you want to hide a better Easter Egg in there? 😄

File: 0002-crypto-conditions/0002-crypto-conditions.md [hunk 4]
```diff
@@ -232,61 +250,77 @@ type
...
 
-A preimage condition is the hash of its own fulfillment. In order to fulfill a preimage condition, a valid preimage must be provided.
+## PREIMAGE {#preimage-feature-suite}
+PREIMAGE refers to conditions that use a preimage as a one-time signature. This feature suite is assigned the feature bit 2^1 = 0x02.
 
-Preimage conditions can be used as a so-called hashlock. Since cryptographically secure hashing functions are preimage-resistant, only the original creator of a preimage condition can feasibly produce the preimage if it contains a large amount of random entropy.
+The fingerprint of a preimage condition is the hash of some arbitrary value. The payload of a preimage fulfillment is the preimage that hashes to the condition's fingerprint. Conditions that use this preimage MUST also rely on a cryptographically secure hashing algorithm. Since cryptographically secure hashing functions are preimage-resistant, only the original creator of a preimage condition can produce the preimage, as long as it contains a large amount of random entropy.
```
Comment (adrianhopebailie) [inline 0002-crypto-conditions/0002-crypto-conditions.md]: It's not obvious from this paragraph that the "arbitrary value" and the payload of the fullfilment are the same thing.

**Comment by adrianhopebailie** [comment]
LGTM (except for inline comments)

File: 0002-crypto-conditions/0002-crypto-conditions.md (hunk 3 above)
Comment (justmoon) [inline 0002-crypto-conditions/0002-crypto-conditions.md]: :+1: for a better easter egg :stuck_out_tongue:

File: 0002-crypto-conditions/0002-crypto-conditions.md [hunk 5]
```diff
@@ -197,13 +199,29 @@ type
...
+
+An example condition in string format:
+
+    cc:0:3:L49pDKQeScuYTG-aUSj-0SdvyvByT_zhtZg6w4ojFUk:35
+
+The example has the following attributes:
+
+| Field                | Value | Description |
+|----------------------|-------|----------------------------------------------|
+| Preface              | `cc`  | Indicates this is a condition. |
```
Comment (justmoon) [inline 0002-crypto-conditions/0002-crypto-conditions.md]: Should be `preface` for consistency.

File: 0002-crypto-conditions/0002-crypto-conditions.md [hunk 6]
```diff
@@ -232,61 +250,77 @@ type
...
+The following is an example fulfillment in string format, for the [example condition](#example-condition):
+
+    cf:0:dGhpcyBpcyBzb21lIHRleHQgdGhhdCBtRHVvMTMgd3JvdGU
+
+
+The example has the following attributes:
+
+| Field                | Value | Description |
+|----------------------|-------|----------------------------------------------|
+| Preface              | `cf`  | Indicates this is a fulfillment. |
```
Comment (justmoon) [inline 0002-crypto-conditions/0002-crypto-conditions.md]: Should be lowercase like the other field names?

File: 0002-crypto-conditions/0002-crypto-conditions.md (hunk 1 above)
Comment (mDuo13) [inline 0002-crypto-conditions/0002-crypto-conditions.md]: I was on the fence about this. I decided it was worthwhile to provide the context about how CCs fit into ILP in one paragraph of the intro. The spec already says it's part of the interledger project, so I think it's important to know why.

That doesn't preclude CCs from being useful elsewhere, of course. Maybe we should add an explicit statement to that effect.

Comment (adrianhopebailie) [inline 0002-crypto-conditions/0002-crypto-conditions.md]: > That doesn't preclude CCs from being useful elsewhere, of course. Maybe we should add an explicit statement to that effect.

👍

**Comment by justmoon** [comment]
LGTM

**Comment by emschwartz** [comment]
@mDuo13 want to merge this PR?

**Comment by mDuo13** [comment]
uh, yes please. I can't

**Pull request merged**